services/core-web/common/actionCreators/projectSummaryActionCreator.js

```diff
@@ -0,0 +1,163 @@
+import { createRequestHeader } from "../utils/RequestHeaders";
+import CustomAxios from "../customAxios";
+
+export const createProjectSummary = ({ mineGuid }, payload) => (dispatch) => {
+  const message = "Successfully created a new project summary";
+  dispatch(request(reducerTypes.CREATE_MINE_PROJECT_SUMMARY));
+  dispatch(showLoading());
+  return CustomAxios()
+    .post(ENVIRONMENT.apiUrl + API.MINE_PROJECT_SUMMARIES(mineGuid), payload, createRequestHeader())
+    .then((response) => {
+      notification.success({ message, duration: 10 });
+      dispatch(success(reducerTypes.CREATE_MINE_PROJECT_SUMMARY));
+      return response;
+    })
+    .catch((err) => {
+      dispatch(error(reducerTypes.CREATE_MINE_PROJECT_SUMMARY));
+      throw new Error(err);
+    })
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const updateProjectSummary = ({ mineGuid, projectSummaryGuid }, payload) => (dispatch) => {
+  dispatch(request(reducerTypes.UPDATE_MINE_PROJECT_SUMMARY));
+  dispatch(showLoading());
+  return CustomAxios()
+    .put(
+      ENVIRONMENT.apiUrl + API.PROJECT_SUMMARY(mineGuid, projectSummaryGuid),
+      payload,
+      createRequestHeader()
+    )
+    .then((response) => {
+      notification.success({
+        message: "Successfully updated project summary",
+        duration: 10,
+      });
+      dispatch(success(reducerTypes.UPDATE_MINE_PROJECT_SUMMARY));
+      return response;
+    })
+    .catch((err) => {
+      dispatch(error(reducerTypes.UPDATE_MINE_PROJECT_SUMMARY));
+      throw new Error(err);
+    })
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const fetchProjectSummariesByMine = ({ mineGuid }) => (dispatch) => {
+  dispatch(request(reducerTypes.GET_PROJECT_SUMMARIES));
+  dispatch(showLoading());
+  return CustomAxios({ errorToastMessage: Strings.ERROR })
+    .get(ENVIRONMENT.apiUrl + API.MINE_PROJECT_SUMMARIES(mineGuid), createRequestHeader())
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_PROJECT_SUMMARIES));
+      dispatch(projectSummaryActions.storeProjectSummaries(response.data));
+    })
+    .catch(() => dispatch(error(reducerTypes.GET_PROJECT_SUMMARIES)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const fetchProjectSummaryById = (mineGuid, projectSummaryGuid) => (dispatch) => {
+  dispatch(request(reducerTypes.GET_PROJECT_SUMMARY));
+  dispatch(showLoading());
+  return CustomAxios({ errorToastMessage: Strings.ERROR })
+    .get(
+      ENVIRONMENT.apiUrl + API.PROJECT_SUMMARY(mineGuid, projectSummaryGuid),
+      createRequestHeader()
+    )
+    .then((response) => {
+      dispatch(success(reducerTypes.GET_PROJECT_SUMMARY));
+      dispatch(projectSummaryActions.storeProjectSummary(response.data));
+    })
+    .catch(() => dispatch(error(reducerTypes.GET_PROJECT_SUMMARY)))
+    .finally(() => dispatch(hideLoading()));
+};
+
+export const addDocumentToProjectSummary = ({ mineGuid, projectSummaryGuid }, payload) => (
+  dispatch
```

Comment:
Did you end up using this? Or are documents being added to the POST/PUT ? I only see this in the tests.

Reply:
I think that may have changed when we went through the document upload stuff and re-worked it, it may have made this redundant, I need to backtrack and make sure of that